From 65e4b3927e9a1d924ab949c1da58d88cb60a521e Mon Sep 17 00:00:00 2001
From: Olivier Dion <olivier.dion@savoirfairelinux.com>
Date: Mon, 26 Jul 2021 09:52:50 -0400
Subject: [PATCH]  pjnath/turn_sock: Fix leak of socket

Opened socket is not closed on binding and quality of service error.  The socket
is owned by the active sock only if everything went well.
---
 pjnath/src/pjnath/turn_sock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pjnath/src/pjnath/turn_sock.c b/pjnath/src/pjnath/turn_sock.c
index dc6304d9f..a6cf34d48 100644
--- a/pjnath/src/pjnath/turn_sock.c
+++ b/pjnath/src/pjnath/turn_sock.c
@@ -1195,6 +1195,7 @@ static void turn_on_state(pj_turn_session *sess,
 					 turn_sock->setting.port_range,
 					 max_bind_retry);
 	    if (status != PJ_SUCCESS) {
+		pj_sock_close(sock);
 		pj_turn_sock_destroy(turn_sock);
 		return;
 	    }
@@ -1205,6 +1206,7 @@ static void turn_on_state(pj_turn_session *sess,
 				    turn_sock->pool->obj_name, NULL);
 	    if (status != PJ_SUCCESS && !turn_sock->setting.qos_ignore_error) 
 	    {
+		pj_sock_close(sock);
 		pj_turn_sock_destroy(turn_sock);
 		return;
 	    }
@@ -1347,6 +1349,7 @@ static void turn_on_state(pj_turn_session *sess,
 #endif
 
 	if (status != PJ_SUCCESS) {
+	    pj_sock_close(sock);
 	    pj_turn_sock_destroy(turn_sock);
 	    return;
 	}
-- 
2.32.0

